scripts/testing/database-emulator-test.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { spawn } = require('child-process-promise');
+
+const EMULATOR_LINK =
+  'https://firebase.google.com/docs/database/downloads/database-emulator.jar';
+const EMULATOR_DIRECTORY = path.join(os.tmpdir(), 'firebase-emulators');
```

Comment:
You can use 'tmp' package for managing temporary directories. It generates a random name and creates the folder for you. 
https://www.npmjs.com/package/tmp

Reply:
Done. Using 'tmp' node module for downloading location now. Eventually we will leverage firebase cli to manage the download and startup of emulators once we have addressed discussion in this [doc](https://docs.google.com/document/d/1k_nmMPL9LRFzxk9yJceNYgHcxgiMeIaAc07YfKUgtqk/edit). But this should be good enough for now.